### PR TITLE
Jz explorethedata styling

### PR DESCRIPTION
**Pull request by LayaTaal**
Updated styling for Explore to Data Charts and legends, including a number of fixes from original pull request.

### CODE REVIEW

File: sass/tji/_variables.scss [hunk 1]
```diff
@@ -17,16 +18,26 @@ $tji-spacing-lg: 2rem;
...
+$tji-line-height-lg: 3.6rem;
+$tji-line-height-xl: 4.6rem;
 
 $tji-chart-legend-size: 1rem;
+
+/* dividers */
+$tji-border: 1px solid #ccc;
+
+/* break-points */
+$tji-break-point-xs: 37.5rem;
```
[the remark this answers is not in the document]
Comment (ifsimicoded) [reply sass/tji/_variables.scss]: having our base font-size be 16px (instead of 10px) makes any em/rem based calculation really complicated. Maybe when we have extra time at some point we want to go back an adjust our base to 10px to make these numbers more intuitive.

File: sass/tji/_charts.scss [hunk 2]
```diff
@@ -1,6 +1,27 @@
 /*--------------------------------------------------------------
 ## CHARTS
 --------------------------------------------------------------*/
+
+.tji-chart {	
+	border: $tji-border;
+	padding: $tji-spacing-md $tji-spacing-sm;
+	margin: $tji-spacing-xs;
```
Comment (ifsimicoded) [inline sass/tji/_charts.scss]: since flexbox-grid classes (col-xs, col-md, etc...) already applies margin styles we shouldn't override these or it will break the grid layout flexbox-grid provides for us. Can we remove this margin style?

You'll see the chart is not centered because of this extra margin that's applied which override flexbox-grid's margin in the screen recording below.
flexboxgrid

File: js/tji-datasets-explore.js [hunk 3]
```diff
@@ -212,7 +215,7 @@ TJIGroupByDoughnutChart.prototype.create_legend = function() {
   var $legend = jQuery('<div class="tji-chart__legend"/>');
   var legend_items = [];
   _.map(keys_sorted, function(key){
-    legend_items.push(jQuery('<div class="tji-chart__legend-item"><span style="background-color:' + colormap[key] + '"></span>' + key + '</div>'));
+    legend_items.push(jQuery('<div class="tji-chart__legend-item"><span style="background-color:' + colormap[key] + '" class="tji-chart__legend-item__item-marker"></span><span class="tji-chart__legend-item__item-description">' + key + '</span></div>'));
```
[the remark this answers is not in the document]
Comment (LayaTaal) [reply js/tji-datasets-explore.js]: Picky indeed! :p but no problem, updated this.

File: sass/tji/_charts.scss [hunk 4]
```diff
@@ -25,23 +46,34 @@
 }
 
 .tji-chart__title {
-  @include font-size(2);
+  @include font-size(1.5);
+  text-align: center;
+  text-transform: capitalize;
+  font-weight: 600;
```
[the remark this answers is not in the document]
Comment (LayaTaal) [reply sass/tji/_charts.scss]: Hm, yes...not sure what happened here but there are supposed to be only two. Thanks for the catch!

File: sass/tji/_charts.scss [hunk 5]
```diff
@@ -25,23 +46,34 @@
...
 
-  span {
-    height: $tji-chart-legend-size;
-    margin-right: $tji-spacing-xs;
-    width: $tji-chart-legend-size;
-  }
+.tji-chart__legend-item__item-marker {
+  height: $tji-chart-legend-size;
+  margin-right: $tji-spacing-xs;
+  width: $tji-chart-legend-size;
```
[the remark this answers is not in the document]
Comment (LayaTaal) [reply sass/tji/_charts.scss]: Yeah, this is a bit weird. It looks like I was having some trouble setting up flexbox and that max-width may have been left behind from testing. But you are right, this is unnecessary.

File: sass/tji/_variables.scss [hunk 6]
```diff
@@ -4,6 +4,7 @@ $primaryBlue: #0B5D93;
 $primaryRed: #CE2727;
 	$secondaryRed: #872729;
 $primaryYellow: #FFFD00;
+$tji-chart-bg: #f5f5f5;
```
[the remark this answers is not in the document]
Comment (LayaTaal) [reply sass/tji/_variables.scss]: Thanks, I like this idea. I just used the same implementation from bootstrap.

File: sass/tji/_variables.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (LayaTaal) [reply sass/tji/_variables.scss]: Thanks Simi, I had not used flexbox grid before and while searching for the breakpoints found an extended version of flexbox grid for sass, which apparently has slightly different breakpoints. I will adjust these to match the original flexbox grid.

Agreed on adjusting the base font-size. I think in general we need to go through following this and improve some practices and consistency.

File: sass/tji/_charts.scss (hunk 2 above)
Comment (LayaTaal) [reply sass/tji/_charts.scss]: I don't see any margin styles in flexbox grid. I think this problem is actually due to in unnecessary margin-right: 2em in _general.scss. I removed that and changed the chart style to only apply the margin on medium+ viewports. This looks good on my end, what do you think?

File: js/tji-datasets-explore.js [hunk 7]
```diff
@@ -476,9 +478,12 @@ TJIChartView.prototype.create_filters = function() {
 
 TJIChartView.prototype.create_filter_checkboxes = function(filter) {
   var that = this;
-  var fieldset = jQuery('<fieldset><legend>' + filter.name.replace(/_/g, " ") + '<i class="fas fa-caret-down"></i></legend></fieldset>');
+  var fieldset = jQuery('<fieldset><legend class="js-chart-legend">' + filter.name.replace(/_/g, " ") + '</legend></fieldset>');
```
Comment (ifsimicoded) [inline js/tji-datasets-explore.js]: Hey @LayaTaal not sure where this code came from? The Filter collapse and expand isn't yet merged.

File: js/tji-datasets-explore.js [hunk 8]
```diff
@@ -476,9 +478,12 @@ TJIChartView.prototype.create_filters = function() {
...
+  var filterset = jQuery('<div class="js-filter-set"><a class="js-toggle-select">Select All</a> / <a class="js-toggle-unselect">Unselect All</a></div>');
```
Comment (ifsimicoded) [inline js/tji-datasets-explore.js]: Same with this, the select-all and deselect-all hasn't yet been merged, so I'm not sure where this code is coming from?

File: sass/tji/_filters.scss [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+>>> EXPLORE THE DATA TABLE OF CONTENTS:
+----------------------------------------------------------------
+# GENERAL
+# FILTERS
+
+/*--------------------------------------------------------------
+## FILTERS
+--------------------------------------------------------------*/
+
+.tji-chart-filters {
```
Comment (ifsimicoded) [inline sass/tji/_filters.scss]: Hey, these look like Jen's styles from a branch that was closed (not merged). Where'd these come from?

File: template-donation.php [hunk 10]
```diff
@@ -0,0 +1,100 @@
+<?php
```
Comment (ifsimicoded) [inline template-donation.php]: Hey also confused where this came from? How do I test this page? Should this be a part of a different Pull Request?

Comment (LayaTaal) [reply template-donation.php]: Haha, man...github is killing me. I updated my branch to the latest master...or so I thought. 

No idea how these got in there, especially a closed branch. 

Apparently I need some hand holding when I pull and merge, =(

**Comment by ifsimicoded** [comment]
@LayaTaal I merged in the updated master (that now includes the filter collapse/expand and de/select all functionality) -- this should have addressed/removed the code bits from jen's former closed branch. Give it a review and make sure your styles are in place as you expect. The only question I think I had left was related to the donation template that got added in. Once that's addressed, merge when rdy.

**Comment by LayaTaal** [comment]
Great, thanks Simi!

Im not at my computer right now, so if you want to pull out the donation form page go for it. It's from another branch on my local computer that must have got pulled in.

File: template-donation.php (hunk 10 above)
Comment (ifsimicoded) [reply template-donation.php]: Hey I fixed up the merge stuff from Jen's old branch. I think the only thing left is the donation template.
